src/main/java/com/cloudbees/jenkins/plugins/amazonecs/ECSService.java

```diff
@@ -243,6 +243,12 @@ TaskDefinition registerTemplate(final ECSCloud cloud, final ECSTaskTemplate temp
                         .withNetworkMode("awsvpc")
                         .withMemory(String.valueOf(template.getMemoryConstraint()))
                         .withCpu(String.valueOf(template.getCpu()));
+                String executionRole = template.getExecutionRole();
+                if(StringUtils.isEmpty(executionRole)){
+                    executionRole = ECSTaskTemplate.DEFAULT_ECS_TASK_EXECUTION_IAM_ROLE_NAME;
```

Comment:
Can you please change it, so that only when a role is defined the executionRole is actually set? Otherwise, people would be surprised by the default value which is not visible to them.

Reply:
Please, also extend the check for existing task definitions to also compare the execution role (see Linke 231)

Reply:
Hi @pgarbe. 
Did that, using default value from UI. The thing is, that usually no one should change the value.
But now I am not "surprising" the end user.

Also added the same check from L231